tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -157,41 +157,19 @@ void test_parse(const CharT* str, const CStringOrStdString& fmt, Parsable& p,
         }
     }
 
-    assert(sstr);
+    return sstr.rdstate();
 }
 
 template <class CharT, class CStringOrStdString, class Parsable>
-void fail_parse(const CharT* str, const CStringOrStdString& fmt, Parsable& p,
+void test_parse(const CharT* str, const CStringOrStdString& fmt, Parsable& p,
     type_identity_t<basic_string<CharT>*> abbrev = nullptr, minutes* offset = nullptr) {
-    p = Parsable{};
-    if (abbrev) {
-        if constexpr (is_same_v<CharT, char>) {
-            *abbrev = "!";
-        } else {
-            *abbrev = L"!";
-        }
-    }
-
-    if (offset) {
-        *offset = minutes::min();
-    }
-
-    basic_stringstream<CharT> sstr{str};
-    if (abbrev) {
-        if (offset) {
-            sstr >> parse(fmt, p, *abbrev, *offset);
-        } else {
-            sstr >> parse(fmt, p, *abbrev);
-        }
-    } else {
-        if (offset) {
-            sstr >> parse(fmt, p, *offset);
-        } else {
-            sstr >> parse(fmt, p);
-        }
-    }
+    assert((parse_state(str, fmt, p, abbrev, offset) & ~ios_base::eofbit) == 0);

```

Comment:
Style question: I see that this assertion and the following one are testing against `0` - would testing against `ios_base::goodbit` be better? Ok to keep as-is if that's preferred.

Reply:
Yes, `goodbit` sounds better. Changed.